tests/src/python/test_qgsvectorlayer.py

```diff
@@ -184,6 +186,25 @@ def dumpEditBuffer(layer):
         print(("%d | %s" % (f.id(), f.geometry().asWkt())))
 
 
+class TestQgsVectorLayerShapefile(unittest.TestCase, FeatureSourceTestCase):
```

Comment:
is thsi test related with this PR?

Reply:
Yes, it's a way to reproduce the original bug. The other vector provider tests all run on the data provider itself, and skip the layer.